### ISSUE TITLE
Styleguide consuming extensive CPU power

### ISSUE THREAD

**Issue by robertrossmann** [atom/styleguide#73]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

When I open the styleguide, it causes CPU usage to spike to 100%-200% of available CPU power and the framerate drops to a sluggish 10-15 fps, causing Atom to feel slow while editing a document with the Styleguide being shown in a separate pane. This occurs even if nothing is being changed in the viewport.

### Steps to Reproduce

1. Open the styleguide (Styleguide -> Show)
1. Observe CPU usage to go up until Styleguide view is closed

**Expected behavior:**

Opening the Styleguide should have minimal impact on performance, at the very least while no elements are being changed (ie. no spinners are on screen)

**Actual behavior:**

Performance drops significantly with Styleguide open, regardless of what is being shown in viewport.

**Reproduces how often:** All the time

### Versions

```
Atom    : 1.34.0-beta0
Electron: 2.0.14
Chrome  : 61.0.3163.100
Node    : 8.9.3

```

### Additional Information

[Performance profile for the Inspector](https://github.com/atom/styleguide/files/2679261/styleguide-perf-profile.json.zip)

> I think this is caused by the fact that the whole Styleguide is contained within a single `render()` function, so that when the moving elements in the styleguide change they cause re-render of the whole styleguide instead of just those parts which really changed.

Separating the Styleguide into small, individual components might help, but I am not a web developer so just throwing ideas here. 🤔

**Comment by simurai** [comment]
Does it help if you "Collapse All" rows?

![screen shot 2018-12-17 at 1 16 06 pm](https://user-images.githubusercontent.com/378023/50066243-004edc80-01fe-11e9-8f7c-c29d1cf11313.png)

Might be better if that _would_ be the default and not render everything.

**Comment by robertrossmann** [comment]
Yup, that drops the CPU usage down to usual levels, although it spikes back up again as soon as I open the _Loading/Progress_ section, even if that's the only section I have visible.
